pkgs/development/libraries/libpwquality/default.nix

```diff
@@ -1,36 +1,47 @@
-{ stdenv, lib, fetchFromGitHub, autoreconfHook, perl, cracklib, python3, fetchpatch }:
+{ stdenv, lib, fetchFromGitHub, autoreconfHook, perl, cracklib, python3, pam }:
 
 stdenv.mkDerivation rec {
   pname = "libpwquality";
-  version = "1.4.2";
+  version = "1.4.4";
+
+  outputs = [ "out" "dev" "lib" "man" ];
 
   src = fetchFromGitHub {
     owner = "libpwquality";
     repo = "libpwquality";
     rev = "${pname}-${version}";
-    sha256 = "0n4pjhm7wfivk0wizggaxq4y4mcxic876wcarjabkp5z9k14y36h";
+    sha256 = "sha256-7gAzrx5VP1fEBwAt6E5zGM8GyuPRR+JxYifYfirY+U8=";
   };
 
   nativeBuildInputs = [ autoreconfHook perl python3 ];
-  buildInputs = [ cracklib ];
-
-  patches = lib.optional stdenv.hostPlatform.isStatic [
-    (fetchpatch {
-      name = "static-build.patch";
-      url = "https://github.com/libpwquality/libpwquality/pull/40.patch";
-      sha256 = "1ypccq437wxwgddd98cvd330jfm7jscdlzlyxgy05g6yzrr68xyk";
-    })
+  buildInputs = [ cracklib ] ++ lib.optionals stdenv.hostPlatform.isLinux [ pam ];
+
+  patches = [
+    # ensure python site-packages goes in $lib output
+    ./python-binding-prefix.patch
   ];
 
-  configureFlags = lib.optional stdenv.hostPlatform.isStatic [
+  configureFlags = lib.optionals stdenv.hostPlatform.isStatic [
     # Python binding generates a shared library which are unavailable with musl build
     "--disable-python-bindings"
   ];
 
   meta = with lib; {
-    description = "Password quality checking and random password generation library";
     homepage = "https://github.com/libpwquality/libpwquality";
-    license = licenses.bsd3;
+    description = "Password quality checking and random password generation library";
+    longDescription = ''
+      The libpwquality library purpose is to provide common functions for
+      password quality checking and also scoring them based on their apparent
+      randomness. The library also provides a function for generating random
+      passwords with good pronounceability. The library supports reading and
+      parsing of a configuration file.
+
+      In the package there are also very simple utilities that use the library
+      function and PAM module that can be used instead of pam_cracklib. The
+      module supports all the options of pam_cracklib.
+    '';
+    license = licenses.gpl2Plus;
```

Comment:
At a glance it looks like it is dual-licensed under `BSD-3-Clause OR GPL-2.0-or-later`. Since Nixpkgs cannot currently express that, I would go with the less restrictive licence of the pair.

Reply:
on most of the rust MIT OR Apache 2 we put it in the comments `/* or */` so I'll do it this way

https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/misc/lscolors/default.nix#L20

Reply:
I do not think that is right. Most systems that check licenses will interpret that as `AND` (e.g. adding an unfree license in the mix would result in the package being considered unfree).

Reply:
There's a lot of packages with `/* or */`

```
λ rg "/\* (or|OR) \*/" . | wc -l
106
```

In this case they're both free. If someone is against GPL or BSD3 they can look at the file and make their decision rather than just getting 1 piece of info.

In the case that something is free or unfree (without support for OR) I'd like to at least know something is licensed unfree and then I can review if it's just optional, or I need to avoid certain files and features, or whatever the conditions are and then either ignore or add an overlay. Typically with free and unfree splits we have an overridable argument that can toggle that feature

Until there's native support for conditionals I much prefer it this way